testkit/src/main/java/org/apache/calcite/sql/parser/SqlParserTest.java

```diff
@@ -2122,6 +2122,29 @@ void checkPeriodPredicate(Checker checker) {
     sql(sql).ok(expected);
   }
 
+  @Test void testGroupByAllOrDistinct() {
+    final String sql = "select deptno from emp\n"
+        + "group by all cube (a, b), rollup (a, b)";
+    final String expected = "SELECT `DEPTNO`\n"
+        + "FROM `EMP`\n"
+        + "GROUP BY CUBE(`A`, `B`), ROLLUP(`A`, `B`)";
+    sql(sql).ok(expected);
+
+    final String sql1 = "select deptno from emp\n"
+        + "group by distinct cube (a, b), rollup (a, b)";
+    final String expected1 = "SELECT `DEPTNO`\n"
```

Comment:
Could you please add some negative case, such as `GROUP BY DISTINCT/ALL ()`?

Reply:
Hi @chunweilei , I find the other tests have no this kind of edge cases, and do you mean `GROUP BY DISTINCT ()` ast should be unparsed as `GROUP BY ()`?

Reply:
What's the behavior of PostgreSQL when meeting `GROUP BY DISTINCT ()`?

Reply:
I tried and that shows `GROUP BY DISTINCT ()` behaves same as `GROUP BY ()`.